### PR TITLE
replace setTint with new setTintInline block showing limited colors

### PR DESCRIPTION
**Pull request by davidsbailey**
fixes https://github.com/code-dot-org/dance-party/issues/240 via "option b" as described in the comments.

depends on https://github.com/code-dot-org/blockly/pull/164

### CODE REVIEW

**Comment by davidsbailey** [comment]
screen shot 2018-11-26 at 9 34 40 am

**Comment by islemaster** [comment]
Do we also need to update the dance party project level?

**Comment by islemaster** [comment]
What is the effect on existing projects?

**Comment by joshlory** [comment]
Is this different from the default palette on https://studio.code.org/projects/artist/?

image

**Comment by davidsbailey** [comment]
Thanks for piling on @islemaster  and @joshlory ! Answers inline:

> Do we also need to update the dance party project level?

Yes, good catch. Will do. Also need to update /s/dance-extras.

> What is the effect on existing projects?

No effect on existing projects. those will keep using the old Dancelab_setTint block.

> Is this different from the default palette on https://studio.code.org/projects/artist/?

Artist is different in that we were restricting the color palette for all places in artist. in dancelab, we want to keep the full color palette for other blocks like "set background". See notes in the bug report on this.

File: apps/src/gamelab/blocks.js [hunk 1]
```diff
@@ -15,6 +15,21 @@ let sprites = () => {
...
+  '#c3b091', // KHAKI
+  '#ffffff', // WHITE
+  '#ff0000', // RED
+  '#ff77ff', // PINK
+  '#ffa000', // ORANGE
+  '#ffff00', // YELLOW
+  '#228b22', // GREEN
+  '#0000cd', // BLUE
+  '#7fffd4', // AQUAMARINE
+  '#843179', // PLUM
```
Comment (joshlory) [inline apps/src/gamelab/blocks.js]: Should these be red/yellow/blue/orange/green/purple?

> only fully saturated primary and secondary colors

**Comment by joshlory** [comment]
Hmm, is the plan to keep the old `Dancelab_setTint` block for compat?  Can we do an in-place upgrade to the new block?

File: apps/src/gamelab/blocks.js (hunk 1 above)
Comment (davidsbailey) [reply apps/src/gamelab/blocks.js]: ooh, yes, I was definitely using the wrong colors! I'm not sure what secondary colors are in this context but I'd assume cyan / magenta not orange / purple:
screen shot 2018-11-26 at 1 44 44 pm

@ryansloan can you please confirm the colors in the left sidebar of the above image look correct?

Comment (davidsbailey) [reply apps/src/gamelab/blocks.js]: Ryan says yes.

**Comment by davidsbailey** [comment]
> Hmm, is the plan to keep the old Dancelab_setTint block for compat? 

Yes.

> Can we do an in-place upgrade to the new block?

We could, but my understanding is that we'd have to modify user code to do this because the color input changing from being a separate block to being inline.

**Comment by davidsbailey** [comment]
This is how the tints look now:
screen shot 2018-11-26 at 2 02 13 pm

**Comment by davidsbailey** [comment]
any other concerns @islemaster @joshlory ? UI tests added here: https://github.com/code-dot-org/code-dot-org/pull/26206/files#diff-9f6495897258945a0beb33d90fde94cbR57

**Comment by davidsbailey** [comment]
added 3 new colors per Marina's input:

screen shot 2018-11-27 at 12 56 10 pm


screen shot 2018-11-27 at 12 47 19 pm

**Pull request merged**